lib/spack/llnl/util/filesystem.py

```diff
@@ -486,29 +497,108 @@ def fix_darwin_install_name(path):
                         stdout=subprocess.PIPE).communicate()[0]
                     break
 
-# Utilities for libraries
 
+def find(root, files, recurse=True):
+    """Search for ``files`` starting from the ``root`` directory.
 
-class LibraryList(collections.Sequence):
-    """Sequence of absolute paths to libraries
+    Like GNU/BSD find but written entirely in Python.
 
-    Provides a few convenience methods to manipulate library paths and get
-    commonly used compiler flags or names
+    Examples:
+
+    .. code-block:: console
+
+       $ find /usr -name python
+
+    is equivalent to:
+
+    >>> find('/usr', 'python')
+
+    .. code-block:: console
+
+       $ find /usr/local/bin -maxdepth 1 -name python
+
+    is equivalent to:
+
+    >>> find('/usr/local/bin', 'python', recurse=False)
+
+    Accepts any glob characters accepted by fnmatch:
+
+    =======  ====================================
+    Pattern  Meaning
+    =======  ====================================
+    *        matches everything
+    ?        matches any single character
+    [seq]    matches any character in ``seq``
+    [!seq]   matches any character not in ``seq``
+    =======  ====================================
+
+    Parameters:
+        root (str): The root directory to start searching from
+        files (str or collections.Sequence): Library name(s) to search for
+        recurse (bool, optional): if False search only root folder,
+            if True descends top-down from the root. Defaults to True.
+
+    Returns:
+        :func:`list`: The files that have been found
     """
+    if isinstance(files, six.string_types):
+        files = [files]
+
+    if recurse:
+        return _find_recursive(root, files)
+    else:
+        return _find_non_recursive(root, files)
+
+
+def _find_recursive(root, search_files):
+    found_files = []
+
+    for path, _, list_files in os.walk(root):
```

Comment:
I think you're completely right, good spot! To be fair, the original method I stole this from did it this way too. I don't think we ever noticed because most of the time the libraries are all in the same directory.